openshift/e2e-tests-openshift.sh

```diff
@@ -0,0 +1,228 @@
+
+  header "Knative Installed successfully"
+}
+
+function deploy_knative_operator(){
+  local COMPONENT="knative-$1"
+
+  cat <<-EOF | oc apply -f -
+	apiVersion: v1
+	kind: Namespace
+	metadata:
+	  name: ${COMPONENT}
+	EOF
+  if oc get crd operatorgroups.operators.coreos.com >/dev/null 2>&1; then
+    cat <<-EOF | oc apply -f -
+	apiVersion: operators.coreos.com/v1
+	kind: OperatorGroup
+	metadata:
+	  name: ${COMPONENT}
+	  namespace: ${COMPONENT}
+	EOF
+  fi
+  cat <<-EOF | oc apply -f -
+	apiVersion: operators.coreos.com/v1alpha1
+	kind: Subscription
+	metadata:
+	  name: ${COMPONENT}-subscription
+	  generateName: ${COMPONENT}-
+	  namespace: ${COMPONENT}
+	spec:
+	  source: ${COMPONENT}-operator
+	  sourceNamespace: $OLM_NAMESPACE
+	  name: ${COMPONENT}-operator
+	  channel: alpha
+	EOF
+  cat <<-EOF | oc apply -f -
+  apiVersion: serving.knative.dev/v1alpha1
+  kind: Install
+  metadata:
+    name: ${COMPONENT}
+    namespace: ${COMPONENT}
+	EOF
+}
+
+
+function enable_knative_interaction_with_registry() {
+  local configmap_name=config-service-ca
+  local cert_name=service-ca.crt
+  local mount_path=/var/run/secrets/kubernetes.io/servicecerts
+
+  oc -n $SERVING_NAMESPACE create configmap $configmap_name
+  oc -n $SERVING_NAMESPACE annotate configmap $configmap_name service.alpha.openshift.io/inject-cabundle="true"
+  wait_until_configmap_contains $SERVING_NAMESPACE $configmap_name $cert_name
+  oc -n $SERVING_NAMESPACE set volume deployment/controller --add --name=service-ca --configmap-name=$configmap_name --mount-path=$mount_path
+  oc -n $SERVING_NAMESPACE set env deployment/controller SSL_CERT_FILE=$mount_path/$cert_name
+}
+
+function create_test_namespace(){
+  oc new-project $TEST_NAMESPACE
+  oc new-project $TEST_NAMESPACE_ALT
+  oc adm policy add-scc-to-user privileged -z default -n $TEST_NAMESPACE
+  oc adm policy add-scc-to-user privileged -z default -n $TEST_NAMESPACE_ALT
+}
+
+function build_knative_client() {
+  failed=0
+  ./hack/build.sh || failed=1
+  return $failed
+}
+
+function run_e2e_tests(){
+  header "Running tests"
+  failed=0
+
+  # adding the basic workflow tests for now
+  # TODO: Link the integration tests written in go here once the PR is merged upstream
+
+  ./kn service create hello --image $KN_DEFAULT_TEST_IMAGE -e TARGET=Knative || failed=1
+  sleep 5
+  ./kn service get || failed=1
+  ./kn service update hello --env TARGET=kn || failed=1
+  sleep 3
+  ./kn revision get || failed=1
+  ./kn service get || failed=1
+  ./kn service create hello --force --image $KN_DEFAULT_TEST_IMAGE -e TARGET=Awesome || failed=1
+  ./kn service create foo --force --image $KN_DEFAULT_TEST_IMAGE -e TARGET=foo || failed=1
+  sleep 5
+  ./kn revision get || failed=1
+  ./kn service get || failed=1
+  ./kn service describe hello || failed=1
+  ./kn service delete hello || failed=1
+  ./kn service delete foo || failed=1
```

Comment:
Hmm, this repetitive `failed=1`  won't help jump to return , we can directly replace all `failed=1` with `return 1` instead, WDYT @rhuss ?

Reply:
We could, otherwise it is often better to collect errors all at once before giving up. That helps to fix multiple errors (if any). Also not sure what happens with the env later, but removing the service might a be not so a bad idea even when the test fails.

So I don't mind much to fall through and return the error at last only

Reply:
okay